lib/trace.c

```diff
@@ -129,6 +129,10 @@ static const char *trace_print_ctx_str[] = {
 	[M0_TRACE_PCTX_FULL]  = "full",
 };
 
+#ifndef __KERNEL__
+M0_INTERNAL bool m0_trace_use_mmapped_buffer(void);
+#endif
```

Comment:
This is not entirely right. Using mmapped buffer and enabling trace should be separate options. An additional interface

m0_trace_disable(bool);
bool m0_trace_is_disabled();

(available in kernel too) is needed.

Besides, this also silently discards error messages, I don't think this is acceptable.

Wouldn't it be better to make M0_TRACE_HIGHEST_ALLOWED_LEVEL a variable that the application can adjust to suppress low priority messages?

Reply:
> Here is `perf top` of writing motr object with `mcp -threads 8` without the patch:
> 
> ![image](https://user-images.githubusercontent.com/18553513/146204739-2a1ddb30-11db-42f4-9b70-c4a0e625204b.png)
> 
> and with the patch:
> 
> ![image](https://user-images.githubusercontent.com/18553513/146204291-8d828f2e-405b-41f8-b354-42cc2bd33dff.png)

@andriytk : Are you seeing the high utilization when only ERROR and above levels of trace are used ,
  Following config level can be set during configure
  --with-trace-max-level=<M0_ALWAYS|M0_FATAL|M0_ERROR|M0_WARN|M0_NOTICE|M0_INFO|M0_DEBUG>
                          set highest trace level that will be enabled, all
                          levels with verbosity above that will not be logged

Reply:
No, I didn't check that. It's very inconvenient to change this at compile time. Note: Hare also has to be re-compiled when Motr is recompiled. This is unpractical.

But I don't think that having only ERRORs will cause any issues. We don't have very much ERROR traces, especially, everything is working fine, do we?

Reply:
> Wouldn't it be better to make M0_TRACE_HIGHEST_ALLOWED_LEVEL a variable that the application can adjust to suppress low priority messages?

Agree. Implemented, see the updated code.